sockeye/inference.py

```diff
@@ -60,12 +65,18 @@ def __init__(self,
                  softmax_temperature: Optional[float] = None,
                  max_output_length_num_stds: int = C.DEFAULT_NUM_STD_MAX_OUTPUT_LENGTH,
                  decoder_return_logit_inputs: bool = False,
-                 cache_output_layer_w_b: bool = False) -> None:
+                 cache_output_layer_w_b: bool = False,
+                 encoder_dtype=np.float32,
+                 decoder_dtype=np.float32) -> None:
         self.model_version = utils.load_version(os.path.join(model_folder, C.VERSION_NAME))
         logger.info("Model version: %s", self.model_version)
         utils.check_version(self.model_version)
 
         config = model.SockeyeModel.load_config(os.path.join(model_folder, C.CONFIG_NAME))
+        config.config_encoder.dtype = encoder_dtype
```

Comment:
why implement this through setters and not in the constructor?

Reply:
+1